reinforcement_learning/trainer/task.py

```diff
@@ -0,0 +1,226 @@
+
+        agent.initialize()
+        global_step = 0
+        for i in xrange(FLAGS.max_episodes):
+
+            s = env.reset()
+            ep_reward = 0
+            ep_steps = 0
+            noises = []
+            actions = []
+            done = False
+
+            while not done:
+                if ep_steps < FLAGS.rand_steps:
+                    action = agent.random_action(s)
+                else:
+                    action, action_org, noise = agent.action_with_noise(s)
+                    noises.append(noise)
+                    actions.append(action_org)
+                action = action.squeeze()
+
+                s2, r, done, info = env.step(action.tolist())
+                ep_reward += r
+                ep_steps += 1
+                global_step += 1
+                agent.store_experience(s, action, r, done, s2)
+
+                # mirror observations and actions
+                flipped_s = util.reverse_obs(s)
+                flipped_s2 = util.reverse_obs(s2)
+                flipped_a = util.reverse_act(action)
+                agent.store_experience(
+                    flipped_s, flipped_a, r, done, flipped_s2)
+
+                agent.train(global_step)
+                s = s2
+
+                if done:
+                    ep_cnt = i + 1
+                    log_metrics(sess,
+                                writer,
+                                summary_ops,
+                                summary_vars,
+                                metrics=(ep_cnt,
+                                         ep_reward,
+                                         ep_steps,
+                                         actions,
+                                         noises))
+                    if ep_cnt % FLAGS.eval_interval == 0:
+                        eval_ep_reward, eval_ep_steps = test(env, agent)
+                        eval_ep_cnt = ep_cnt / FLAGS.eval_interval
+                        log_metrics(sess,
+                                    writer,
+                                    eval_summary_ops,
+                                    eval_summary_vars,
+                                    metrics=(eval_ep_cnt,
+                                             eval_ep_reward,
+                                             eval_ep_steps,
+                                             None,
+                                             None),
+                                    test=True)
+                        ckpt_path = saver.save(sess,
+                                               model_path,
+                                               global_step=global_step)
+                        tf.logging.info('Model saved to {}'.format(ckpt_path))
+    env.close()
+
+
+def main(argv):
+    del argv
+    for k, v in FLAGS.flag_values_dict().iteritems():
+        tf.logging.info('{}: {}'.format(k, v))
+    train()
+
+
+if __name__ == '__main__':
+    tf.logging.set_verbosity(tf.logging.DEBUG)
+    app.run(main)
```

Comment:
parse the args in this `if` clause and just call `main(args)` instead.

`app.run` causes issues if running from inside a notebook, for example.

Reply:
fixed in 994e1c88e71fa20ca85eb6ed2b81e816f8e5c475